File: src/main/java/com/godwin/jsonparser/ui/IParserWidget.java
package com.godwin.jsonparser.ui;

import javax.swing.*;

/**
 * Created by Godwin on 4/21/2018 12:32 PM for json.
 *
 * @author : Godwin Joseph Kurinjikattu
 */
public interface IParserWidget {
    void createParserSession();

    void closeCurrentParserSession();

    int getTabCount();

    JComponent getComponent();
}

File: src/main/kotlin/com/godwin/jsonparser/ui/IParserWidget.kt
package com.godwin.jsonparser.ui

import com.godwin.jsonparser.ui.tabs.IParserTabs
import javax.swing.JComponent

interface IParserWidget {
    fun createParserSession()
    fun createParserSessionWithContent(json: String)
    fun closeCurrentParserSession()
    fun getTabCount(): Int
    fun getTabs(): IParserTabs?
    fun getComponent(): JComponent
}
